core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -206,6 +213,15 @@ public function findJoinCriteriasForTables(LogTable $logTable, $availableLogTabl
 
                 break;
             }
+
+            $otherJoins = $logTable->getWaysToJoinToOtherLogTables();
+            foreach ($otherJoins as $joinTable => $column) {
+                if($availableLogTable->getName() == $joinTable) {
+                    $join = sprintf("%s.%s = %s.%s", $table, $column, $availableLogTable->getName(), $column);
```

Comment:
I'm not 100% sure it looks like this is also used in line 131:

```

                        $table['joinOn'] = $this->findJoinCriteriasForTables($logTable, $availableLogTables);
``` 

when dealing with arrays. but in the above code  in `addMissingTablesNeededForJoins()` we don't add missing tables when it is an array? not sure if you know what I mean, or if it's an issue ... but may need to be looked at and probably needs a test. currently we're adding only needed tables when table is defined as a string `log_xyz` but not through an array `array('table' => 'log_xyz')`.